ibis/backends/pandas/__init__.py

```diff
@@ -212,6 +212,21 @@ def to_pyarrow(
         else:
             return pa.scalar(output)
 
+    def to_pyarrow_batches(
+        self,
+        expr: ir.Expr,
+        *,
+        params: Mapping[ir.Scalar, Any] | None = None,
+        limit: int | str | None = None,
+        chunk_size: int = 1000000,
+        **kwargs: Any,
+    ) -> pa.ipc.RecordBatchReader:
+        pa = self._import_pyarrow()
+        pa_table = self.to_pyarrow(expr, params=params, limit=limit)
+        return pa.RecordBatchReader.from_batches(
```

Comment:
How about creating record batches directly from the dataframe using `pa.RecordBatch.from_pandas` and explicitly raising here if the result is not table-like? Do we support chunked arrays here as well?

Reply:
I think this requires all the same acrobatics as the `to_pyarrow` handler, since you can't create a `RecordBatch` from a series, so we'd need to promote single-column result sets to single-column tables (at which point we might as well just use the `to_pyarrow` to feed the batches)